scripts/audit-spark-3.2.sh

```diff
@@ -0,0 +1,101 @@
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+# This script generates the commits that went in Apache Spark for audit.
+# Audit is required to evaluate if the code needs to be updated based
+# on new commits merged in Apache Spark. This currently audits changes for
+# Spark-3.2 (master branch).
+# Arguments:
+#   lastcommit - File which contains the latest commit hash when this script ran last.
+#   basebranch - branch in Apache Spark for which commits needs to be audited.
+#                Currently it's master as Spark-3.2 branch is not cut yet.
+
+
+set -ex
+
+lastcommit=""
+basebranch="master"
+REF=${REF:-"main"}
+REF=main
+while getopts v:a: flag
+do
+  case "${flag}" in
+      v) lastcommit=${OPTARG};;
+      b) basebranch=${OPTARG};;
+  esac
+done
+
+SPARK_TREE="$HOME/spark-3.2/spark"
+rm -rf $SPARK_TREE && git clone https://github.com/apache/spark.git $SPARK_TREE
+
+if [ -f "$lastcommit" ]; then
+    cd ${SPARK_TREE}
+    latestcommit=`cat ${lastcommit}`
+    git checkout $basebranch
+    git log --oneline HEAD...$latestcommit -- sql/core/src/main sql/catalyst/src/main | tee $COMMIT_DIFF_LOG
+    git log HEAD -n 1 --pretty="%h" > ${lastcommit}
+
+    cd $WORKSPACE
+    COMMIT_UPDATE=`git diff ${lastcommit}`
+    if [ -n "$COMMIT_UPDATE" ]; then
+        git config --global user.name blossom
+        git config --global user.email blossom@nvidia.com
+        git add ${lastcommit}
+        git commit -m "Update latest commit-id for org.apache.spark branch ${basebranch}"
+        git push origin HEAD:$REF
+    else
+        echo "No commit update"
+    fi
+else
+  ## Below sequence of commands were used to get the initial list of commits to audit branch-3.2-SNAPSHOT(which is currently `master` branch)
+  ## It filters out all the commits that were audited until 3.1.1-rc3.
+  ## There wasn't easy way to get the list of commits to audit for branch-3.2-SNAPSHOT.
+  ## Spark release works in this way -  Once the release branch is cut, PR's are merged into master and then cherry-picked to release branches.
+  ## This causes different commit ids for the same PR in different branches(master & release branch).
+  ## We need to find the common parent before branch-3.1 was cut. In this case commit id 990bee9c58e is the one.
+  ## So we get all commits from master and branch-3.1 until the common parent commit and then filter it based on commit header message i.e
+  ## if the commit header is same, it means it is cherry-picked to branch-3.1(implying that commit is already audited).
+  echo "file $lastcommit not found"
+  cd ${SPARK_TREE}
+
+  ## Get all the commits from TOT tagv3.1.1-rc3 to 990bee9c58e
+  git checkout v3.1.1-rc3
```

Comment:
This can be a command line option

Reply:
Thanks! changed it to command line option.